processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt

```diff
@@ -233,7 +230,24 @@ class DataCompatProcessor(
                         propertyMap.keys.joinToString(
                             prefix = "return ",
                             separator = "·&& ",
-                            transform = { "$it·==·other.$it" },
+                            transform = {
+                                val resolvedType = it.type.resolve()
+                                val isFloat =
+                                    resolvedType.toClassName() == Float::class.asTypeName()
+                                val isDouble =
+                                    resolvedType.toClassName() == Double::class.asTypeName()
+                                if (isFloat || isDouble) {
+                                    if (resolvedType.isMarkedNullable && isDouble) {
```

Comment:
nit:
replace with `with(it) { ...}`

Reply:
done (if I understood the suggestion correctly)